connectors/yarpc/yarpc.go

```diff
@@ -38,11 +39,43 @@ import (
 )
 
 const (
-	_defaultServiceName        = "dosa-gateway"
-	errCodeNotFound      int32 = 404
-	errCodeAlreadyExists int32 = 409
+	_defaultServiceName         = "dosa-gateway"
+	errCodeNotFound      int32  = 404
+	errCodeAlreadyExists int32  = 409
+	errInvalidHandler    string = "no handler for service"
+	errConnectionRefused string = "getsockopt: connection refused"
 )
 
+// ErrInvalidHandler is used to help deliver a better error message when
+// users have misconfigured the yarpc connector
+type ErrInvalidHandler struct{}
+
+// Error implements the error interface
+func (*ErrInvalidHandler) Error() string {
+	return "the service name for the gateway is wrong, make sure your configuration is correct for your environment"
```

Comment:
The gateway (%q) refused to handle the configured scope %q; this is usually due to a mismatch between gateway and scope in your configuration.